grove/measurements/estimation.py

```diff
@@ -0,0 +1,170 @@
+
+    :param PauliSum pauli_sums: The Pauli sum to process.
+    :return: a purely hermitian Pauli sum.
+    :rtype: PauliSum
+    """
+    if not isinstance(pauli_sums, PauliSum):
+        raise TypeError("not a pauli sum. please give me one")
+    new_term = sI(0) * 0.0
+    for term in pauli_sums:
+        new_term += term_with_coeff(term, term.coefficient.real)
+
+    return new_term
+
+
+def get_rotation_program(pauli_term):
+    """
+    Generate a rotation program so that the pauli term is diagonal
+
+    :param PauliTerm pauli_term: The Pauli term used to generate diagonalizing
+                                 one-qubit rotations.
+    :return: The rotation program.
+    :rtype: Program
+    """
+    meas_basis_change = Program()
+    for index, gate in pauli_term:
+        if gate == 'X':
+            meas_basis_change.inst(RY(-np.pi / 2, index))
+        elif gate == 'Y':
+            meas_basis_change.inst(RX(np.pi / 2, index))
+        elif gate == 'Z':
+            pass
+        else:
+            raise ValueError()
+
+    return meas_basis_change
+
+
+def get_parity(pauli_terms, bitstring_results):
+    """
+    Calculate the eigenvalues of Pauli operators given results of projective measurements
+
+    The single-qubit projective measurement results (elements of
+    `bitstring_results`) are listed in physical-qubit-label numerical order.
+
+    An example:
+
+    Consider a Pauli term Z1 Z5 Z6 I7 and a collection of single-qubit
+    measurement results corresponding to measurements in the z-basis on qubits
+    {1, 5, 6, 7}. Each element of bitstring_results is an element of
+    :math:`\{0, 1\}^{\otimes 4}`.  If [0, 0, 1, 0] and [1, 0, 1, 1]
+    are the two projective measurement results in `bitstring_results` then
+    this method returns a 1 x 2 numpy array with values [[-1, 1]]
+
+    :param List pauli_terms: A list of Pauli terms operators to use
+    :param bitstring_results:
+    :return:
+    """
+    max_weight_pauli_index = np.argmax(
+        [len(term.get_qubits()) for term in pauli_terms])
+    active_qubit_indices = sorted(
+        pauli_terms[max_weight_pauli_index]._ops.keys())
+    index_mapper = dict(zip(active_qubit_indices,
+                            range(len(active_qubit_indices))))
+
+    results = np.zeros((len(pauli_terms), len(bitstring_results)))
+
+    # convert to array so we can fancy index into it later.
+    # list() is needed to cast because we don't really want a map object
+    bitstring_results = list(map(np.array, bitstring_results))
+    for row_idx, term in enumerate(pauli_terms):
+        memory_index = np.array(list(map(lambda x: index_mapper[x],
+                                         sorted(term.get_qubits()))))
+
+        results[row_idx, :] = [-2 * (sum(x[memory_index]) % 2) +
+                               1 for x in bitstring_results]
+    return results
+
+
+def estimate_pauli_set(pauli_terms, basis_transform_dict, program,
```

Comment:
How about we name this `estimate_pauli_sum` and we allow `pauli_terms` to be either a list of terms or a `PauliSum` but we immediately convert it to a list as `pauli_terms = list(pauli_terms)`?
This way one could pass a more intuitive object (the sum) but if one wants precise control over the ordering to make sense of the covariance matrix elements they can also pass a list ordered according to their liking.